File: Android/app/src/main/java/com/dreamproject/inwords/presentation/viewScenario/authorisation/registration/RegistrationViewModel.java
package com.dreamproject.inwords.presentation.viewScenario.authorisation.registration;

import com.dreamproject.inwords.data.dto.UserCredentials;
import com.dreamproject.inwords.domain.interactor.authorisation.AuthorisationInteractor;
import com.dreamproject.inwords.presentation.viewScenario.authorisation.AuthorisationViewModel;

import io.reactivex.Completable;

public class RegistrationViewModel extends AuthorisationViewModel {

    RegistrationViewModel(AuthorisationInteractor authorisationInteractor) {
        super(authorisationInteractor);
    }

    @Override
    protected Completable performAuthAction(UserCredentials userCredentials) {
        return authorisationInteractor.signUp(userCredentials);
    }
}

File: Android/app/src/main/java/ru/inwords/inwords/presentation/view_scenario/authorisation/registration/RegistrationViewModel.kt
package ru.inwords.inwords.presentation.view_scenario.authorisation.registration

import io.reactivex.Completable
import ru.inwords.inwords.domain.interactor.authorisation.AuthorisationInteractor
import ru.inwords.inwords.presentation.view_scenario.authorisation.AuthorisationViewModel
import ru.inwords.inwords.profile.data.bean.UserCredentials

class RegistrationViewModel(authorisationInteractor: AuthorisationInteractor) : AuthorisationViewModel(authorisationInteractor) {
    override fun performAuthAction(userCredentials: UserCredentials): Completable {
        return authorisationInteractor.signUp(userCredentials)
    }
}